0014-fungible-token-standard.md

```diff
@@ -2,85 +2,178 @@
+
+#### The Admin Contract
+```TypeScript
+export type FungibleTokenAdminBase = SmartContract & {
+  canMint(accountUpdate: AccountUpdate): Promise<Bool>
+  canChangeAdmin(admin: PublicKey): Promise<Bool>
+  canPause(): Promise<Bool>
+  canResume(): Promise<Bool>
 }
 ```
 
-### Viewable
-For retrieving information about the token, such as the total or circulating supply, account balances, etc., we define the `Viewable` interface.
+An admin contract needs to provide the methods defined in `FungibleTokenAdminBase`, thus implementing that interface. Each of the methods will be called by the token contract to check permissions, and will return a `Bool` value to grant or deny permission.
+
+An example implementation that allows a priviledged key to perform any kind of administrative action is provided with the standard implementation.
 
+#### Deploying the Contract
 ```TypeScript
-interface Viewable {
-  getAccountOf: (address: PublicKey) => ReturnType<typeof Account>;
-  getBalanceOf: (address: PublicKey, options: ViewableOptions) => UInt64;
-  getTotalSupply: (options: ViewableOptions) => UInt64;
-  getCirculatingSupply: (options: ViewableOptions) => UInt64;
-  getDecimals: () => UInt64;
+export interface FungibleTokenDeployProps extends Exclude<DeployArgs, undefined> {
+  /** Address of the contract controlling permissions for administrative actions */
+  admin: PublicKey
+  /** The token symbol. */
+  symbol: string
+  /** A source code reference, which is placed within the `zkappUri` of the contract account. */
+  src: string
+  /** Number of decimals in a unit */
+  decimals: UInt8
 }
+
+async deploy(props: FungibleTokenDeployProps)
+```
+
+Deploys the token contract. The token admin contract is assumed to be already deployed, at the address given by `admin`. The token symbol, number of digits, and reference to the source code are to be provided.
+
+#### Changing the Admin Contract
+```TypeScript
+@method async setAdmin(admin: PublicKey)
+```
+
+#### Minting Tokens
+```TypeScript
+@method.returns(AccountUpdate) async mint(recipient: PublicKey, amount: UInt64)
+```
+
+Creates `amount` new tokens in the token account of `recipient`.
+
+Requires `canMint()` of the admin contract to return `Bool(true)`.
+
+#### Pausing and Resuming Transfers
+```TypeScript
+@method async pause()
+@method async resume()
+```
+
+After `pause()` has been successfully called, users will not be able to move or burn tokens, until `resume()` has been called successfully.
+
+Those methods call `canPause()` and `canResume()` of the admin contract, respectively, and only succeed on `Bool(true)`.
+
+### Events
+
+Minting, burning, or transferring tokens (via the `transfer()` method, not via approving account updates) will emit the following events:
+
+```TypeScript
+export class MintEvent extends Struct({
```

Comment:
export needed?

Reply:
Isn't it useful to have them exported in order to allow third parties to analyse the events?